main/test/ca/uwaterloo/flix/language/phase/TestResolver.scala

```diff
@@ -1397,4 +1397,37 @@ class TestResolver extends FunSuite with TestUtils {
     val result = compile(input, Options.TestWithLibMin)
     expectError[ResolutionError.IllegalWildType](result)
   }
+
+  test("IllegalForEachYield.01") {
```

Comment:
Usually the naming convention is something like "XYZError.SpecificCase.Number".

So something like `UndefinedName.ForYield.01` would be nice here